random-fu/src/Data/Random/Distribution/Normal.hs

```diff
@@ -159,18 +160,32 @@ doubleStdNormalV = 2.4567663515413507e-3
 
 {-# NOINLINE doubleStdNormalZ #-}
 doubleStdNormalZ :: Ziggurat UV.Vector Double
-doubleStdNormalZ = mkZiggurat_ True 
-        normalF normalFInv 
-        doubleStdNormalC doubleStdNormalR doubleStdNormalV 
+doubleStdNormalZ = mkZiggurat_ True
+        normalF normalFInv
+        doubleStdNormalC doubleStdNormalR doubleStdNormalV
         getIU
         (normalTail doubleStdNormalR)
-    where 
+    where
         getIU :: RVarT m (Int, Double)
         getIU = do
-            !w <- getRandomWord64
+            !w <- Random.uniformWord64 RGen
             let (u,i) = wordToDoubleWithExcess w
             return $! (fromIntegral i .&. (doubleStdNormalC-1), u+u-1)
 
+-- NOTE: inlined from random-source
```

Comment:
This is interesting - I should check what random itself does.

Reply:
That was another one that required delving into actual math part of number generation, which I wanted to leave for you to figure out.

Reply:
I haven't checked fully but I think is actually needed for the Ziggurat algorithm - see https://en.wikipedia.org/wiki/Ziggurat_algorithm and possibly look in mwc-random (although that uses a modified Ziggurat algorithm).